### PR TITLE
Fix for reflection issue with renderer selection

### PR DESCRIPTION
**Pull request by kaspersorensen**
Updated ReflectionUtils.getHierarchyDistance() to support mix of class and interface hierarchy distance.

### CODE REVIEW

File: engine/core/src/main/java/org/datacleaner/util/ReflectionUtils.java [hunk 1]
```diff
@@ -331,54 +331,80 @@ public static boolean isWildcard(Type type) {
...
         }
     }
 
-    private static int getInterfaceHierarchyDistance(Class<?> subtype, Class<?> supertype) {
+    private static int getClassHierarchyDistance(final Class<?> subtype, final Class<?> supertype) {
         if (subtype == supertype) {
             return 0;
         }
+        if (subtype == Object.class) {
+            throw new IllegalArgumentException("Finished traversing hierarchy and found Object.class");
```
Comment (LosD) [inline engine/core/src/main/java/org/datacleaner/util/ReflectionUtils.java]: I might be wrong, but this seems like a bad use of exceptions to me; judging by the code below (wrapping them in empty catch blocks), there is nothing exceptional about this, it's just a "not found".

**Comment by LosD** [comment]
Except what looks like an exception-based integer version of the [tri-state boolean](http://thedailywtf.com/articles/What_Is_Truth_0x3f_), it looks good to me :)

File: engine/core/src/main/java/org/datacleaner/util/ReflectionUtils.java (hunk 1 above)
Comment (kaspersorensen) [inline engine/core/src/main/java/org/datacleaner/util/ReflectionUtils.java]: Yeah it is a case of "we tried to traverse through the hierarchy but didn't fint the right way this time" situation. How would you rather return? We could return -1 instead or something, but isn't that excactly the tri-state boolean? :)

Comment (LosD) [inline engine/core/src/main/java/org/datacleaner/util/ReflectionUtils.java]: Hehe, yeah maybe, but it's still better than using an unexceptional exception :)

I'd maybe consider Integer.MAX, as that is the closest thing to "infinite distance", but -1 probably also works fine (and would be 0xFFFFFFFFF, so looking at it as unsigned, it is even further away :)).

**Comment by kaspersorensen** [comment]
Fixed it and merging :)

**Pull request merged**